src/DotLiquid/Block.cs

```diff
@@ -31,8 +28,7 @@ protected override void Parse(List<string> tokens)
             string token;
             while ((token = tokens.Shift()) != null)
             {
-                Match isTagMatch = IsTag.Match(token);
-                if (isTagMatch.Success)
+                if (token.Length > 4 && token.Substring(0, 2) == Tokenizer.TagStart)
```

Comment:
Generally C# code analyzers frown upon the use of string equality operator and want instead to use the explicit string.Equal with a culture specifier such as ordinal case sensitive (or not)

Reply:
might be even better to just compare token[0] and token[1] as chars. 
```
if (token.Length > 4 && token[0] == Tokenizer.CharCurlyBraceStart && token[1] == Tokenizer.CharPercent)
```